File: java/pattern/src/factory/abstractfactory/LowFactory.java
package factory.abstractfactory;

/**
 * 低端汽车
 */
public class LowFactory implements CarFactory {

    @Override
    public Engine createEngine() {
        return new LowEngine();
    }

    @Override
    public Seat createSeat() {
        return new LowSeat();
    }

    @Override
    public Tyre creatTyre() {
        return new LowTyre();
    }
}

File: php/factory/abstractfactory/LowFactory.php
<?php

// include 'CarFactory.php';

// include 'Engine.php';
// include 'Seat.php';
// include 'Tyre.php';

class LowFactory implements CarFactory
{
    public function createEngine()
    {
        return new LowEngine();
    }

    public function createSeat()
    {
        return new LowSeat();
    }

    public function createTyre()
    {
        return new LowTyre();
    }
}

